assets/js/app.js

```diff
@@ -0,0 +1,423 @@
+"use strict";
+
+const CONFIG = {
+  MAX_SELECTED_COINS: 5,
+  CACHE_TIMEOUT: 2 * 60 * 1000,
+  CHART_UPDATE_INTERVAL: 2000,
+  CHART_MAX_POINTS: 10,
+  COLORS: ["red", "blue", "green", "orange", "purple"],
+  LOADING_HTML: `
+    <div class="text-center p-5">
+      <div class="spinner-border text-primary" role="status">
+        <span class="visually-hidden">Loading...</span>
+      </div>
+      <p class="mt-2">Loading coins...</p>
+    </div>
+  `,
+  API_ENDPOINTS: {
+    COINS_LIST: "https://api.coingecko.com/api/v3/coins/list",
+    MARKETS: "https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1",
+    PRICES: "https://min-api.cryptocompare.com/data/pricemulti",
+    COIN_INFO: (id) => `https://api.coingecko.com/api/v3/coins/${id}`
+  }
+};
+
+const selectedCoins = new Set();
+let chartInstance = null;
+
+const saveToLocalStorage = (key, data) => {
```

Comment:
didn't dig deeply but looked like you messed several entities here. the selected coins are somehow reflected in the reports page, but they don't show as switched on when navigating back to the coins page